src/components/pages/index.js

```diff
@@ -22,6 +22,20 @@ function Index({ initialState }) {
   const currentHomeLanguage = homeLanguages[currentHomeLanguageIndex];
   const [isSearching, setIsSearching] = useState(false);
 
+  const handlePageShow = event => {
+    if (event.persisted) {
+      window.location.reload(true);
```

Comment:
This would reset everything, don't we want to keep what the user typed in? You can do that by setIsSearching(false). This way I guess we do a round trip to the server.